### PR TITLE
docs: Add parquet_opendal in related projects

### PR DESCRIPTION
**Pull request by Xuanwo**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-rs/issues/6235

# Rationale for this change
 

Covered in https://github.com/apache/arrow-rs/issues/6235

# What changes are included in this PR?

Only docs change.

# Are there any user-facing changes?

No.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,12 +92,15 @@ There are several related crates in different repositories
...
 | [`object_store_opendal`] | Use [`opendal`] as [`object_store`] backend | [(README)][object_store_opendal-readme] |
+| [`parquet_opendal`]      | Use [`opendal`] for [`parquet`] Arrow IO    | [(README)][parquet_opendal-readme]      |
 
 [`datafusion`]: https://crates.io/crates/datafusion
 [`ballista`]: https://crates.io/crates/ballista
 [`object_store_opendal`]: https://crates.io/crates/object_store_opendal
 [`opendal`]: https://crates.io/crates/opendal
 [object_store_opendal-readme]: https://github.com/apache/opendal/blob/main/integrations/object_store/README.md
+[`parquet_opendal`]: https://crates.io/crates/parquet_opendal
+[parquet_opendal-readme]: https://github.com/apache/opendal/blob/main/integrations/parquet/README.md
```
Comment (flaneur2020) [inline README.md:103]: are all these project names expected to be wrapped with ` ? 👀

Comment (Xuanwo) [reply README.md:103]: Yes, I believe the motivation is to display the crate name as a code block.

image

**Comment by tustvold** [comment]
I'm curious if you have any concrete numbers to share comparing this to using object_store_opendal? I'm just thinking that it isn't very obvious why you would use one over the other.

_I'm also a little lukewarm about maintaining a list of downstream projects, at least historically this was a list of projects that shared the arrow PMC. DataFusion has now split off, but still has non-trivial overlap..._

**Comment by alamb** [comment]
> I'm also a little lukewarm about maintaining a list of downstream projects, at least historically this was a list of projects that shared the arrow PMC. DataFusion has now split off, but still has non-trivial overlap...

I think as long as the community maintains the list of downstream projects it is a good thing to have in this crate as it helps the overall user community potentially find the software that might be relevant to them.

**Review by alamb — APPROVED** [review]
Thank you

**Comment by alamb** [comment]
> I'm curious if you have any concrete numbers to share comparing this to using object_store_opendal? I'm just thinking that it isn't very obvious why you would use one over the other.

One reason would be that you already use openDAL in your project

Another would be that you use a data source that is supported by open DAL (e.g. FTP) but not by object_store

**Comment by tustvold** [comment]
> One reason would be that you already use openDAL in your project

Right my question was why you would not use object_store_opendal, and the first-party support for object_store. As it stands we're now linking to two different ways to achieve the same end, use parquet with opendal, and this seems perhaps redundant?

**Comment by alamb** [comment]
> > One reason would be that you already use openDAL in your project
> 
> Right my question was why you would not use object_store_opendal with the first-party support for object_store. As it stands we're now linking to two different ways to achieve the same end, use parquet with opendal, and this seems perhaps redundant?

If I were a user without knowledge of how parquet / object_store / arrow-rs / open_dal were all connected, I think it would be simpler for me to pattern match:

"I use opendal in my system. I want to read parquet from opendal. Thus I should use parquet_opendal"

Otherwise I have to know to use a level of indirection "I should use object_store_opendal to adapt my opendal data source to an ObjectStore and then use parquet"

What I think actually makes more sense is to avoid the "first party" distinction somehow

**Comment by tustvold** [comment]
Fair enough, I just find the story of having a single unified IO abstraction for the ecosystem quite compelling, i.e. the model of object_store_opendal, and was curious if there are issues with the interface that justify skipping this abstraction. I agree people using OpenDAL will pattern match and use parquet_opendal, but if object_store_opendal would perform just the same this seems like a missed opportunity for broader ecosystem interoperability.

Anyway I don't feel strongly, I'm just interested if we can improve the ObjectStore interface

**Comment by alamb** [comment]
Thanks again!

**Pull request merged**